lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -1044,44 +1047,45 @@ def get(self, request, subsection_id):
                 developer_message='Invalid UserID',
                 error_code='invalid_user_id'
             )
-
-        try:
-            original_grade = PersistentSubsectionGrade.read_grade(user_id, usage_key)
-        except PersistentSubsectionGrade.DoesNotExist:
-            results = SubsectionGradeResponseSerializer({
-                'original_grade': None,
-                'override': None,
-                'history': [],
-                'subsection_id': usage_key,
-                'user_id': user_id,
-                'course_id': None,
-            })
-
-            return Response(results.data)
-
-        limit_history_request_value = request.GET.get('history_record_limit')
-        if limit_history_request_value is not None:
+        override = None
+        history = []
+        history_record_limit = request.GET.get('history_record_limit')
+        if history_record_limit is not None:
             try:
-                history_record_limit = int(limit_history_request_value)
+                history_record_limit = int(history_record_limit)
             except ValueError:
                 history_record_limit = 0
 
         try:
-            override = original_grade.override
-            if limit_history_request_value is not None:
-                history = reversed(list(override.history.all().order_by('-history_date')[:history_record_limit]))
-            else:
-                history = override.history.all().order_by('history_date')
-        except PersistentSubsectionGradeOverride.DoesNotExist:
-            override = None
-            history = []
+            original_grade = PersistentSubsectionGrade.read_grade(user_id, usage_key)
+            if original_grade is not None and hasattr(original_grade, 'override'):
+                override = original_grade.override
+                history = list(original_grade.override.history.all().order_by('history_date')[:history_record_limit])
+            grade_data = {
+                'earned_all': original_grade.earned_all,
+                'possible_all': original_grade.possible_all,
+                'earned_graded': original_grade.earned_graded,
+                'possible_graded': original_grade.possible_graded,
+            }
+
+        except PersistentSubsectionGrade.DoesNotExist:
+            student = get_user_model().objects.get(id=user_id)
+            course_structure = get_course_blocks(student, usage_key)
+            subsection_grade_factory = SubsectionGradeFactory(student, course_structure=course_structure)
+            grade = subsection_grade_factory.create(course_structure[usage_key], read_only=True, force_calculate=True)
+            grade_data = {
```

Comment:
This block of assignment is repeated. Can you make this assignment just done once?

Reply:
grade_data is populated from different objects.

Reply:
@schenedx it's similar looking, but not identical.  In the `try` block, we're getting data from a `PersistentSubsectionGrade` object, which has `earned_all`, etc. as direct attributes.  In the `except` block, we get data from a `CreateSubsectionGrade` object, which stores those values inside `all_total` and `graded_total` attributes.